src/core/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,26 @@
+import IconButton from '@mui/material/IconButton';
+import { MouseEvent } from 'react';
+import './IconButton.css';
+
+interface CustomIconButtonProps {
+  icon: JSX.Element;
+  label?: string;
+  variant?: 'primary' | 'secondary';
+  handleClick: (event: MouseEvent<HTMLButtonElement>) => void;
+}
+
+export const CustomIconButton = ({ icon, label, variant = 'primary', handleClick }: CustomIconButtonProps) => {
+  const validVariants = ['primary', 'secondary'];
+  const defaultVariant = 'primary';
+
+  variant = validVariants.includes(variant) ? variant : defaultVariant;
+
+  return (
+    <>
+      <IconButton onClick={handleClick} className={`icon-button icon-button--${variant}`}>
```

Comment:
lets add `icon-button--${variant}`` only if the variant's value exists, otherwise we should hide this class. The styles by it self will apply correctly to the corresponding classes.  Invalid classes will be ignored.